### PR TITLE
Add tests for upgrade scenario

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

File: test/definitions/scenarios/capsule_upgrade_test.rb [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+      )
+    end
+  end
+
+  describe Scenarios::Satellite::Migrations do
+    let(:scenario) do
+      Scenarios::Satellite::Migrations.new
+    end
+
+    it 'composes all steps for Capsule on EL8' do
```
Comment (evgeni) [inline test/definitions/scenarios/capsule_upgrade_test.rb:92]: When I wrote this, I only looked at "enable modules", but in reality there are more steps:
```
        add_step(Procedures::Repositories::Setup.new(:version => target_version))
        if el8?
          modules_to_switch = ['postgresql:13']
          add_step(Procedures::Packages::SwitchModules.new(:module_names => modules_to_switch))
          modules_to_enable = ["#{feature(:instance).downstream.module_name}:#{el_short_name}"]
          add_step(Procedures::Packages::EnableModules.new(:module_names => modules_to_enable))
        end
        add_step(Procedures::Packages::Update.new(
          :assumeyes => true,
          :download_only => true
        ))
        add_step(Procedures::Service::Stop.new)
        add_step(Procedures::Packages::Update.new(:assumeyes => true, :clean_cache => false))
        add_step_with_context(Procedures::Installer::Run)
        add_step(Procedures::Installer::UpgradeRakeTask)
```

mind adding these?

**Review by evgeni — CHANGES_REQUESTED** [review]
ACK once the "Migrations" part is properly filled in (which I avoided the last time, SORRY!)

File: test/definitions/scenarios/capsule_upgrade_test.rb (hunk 1 above)
Comment (ehelms) [reply test/definitions/scenarios/capsule_upgrade_test.rb:92]: Doh - I added these in the Satellite tests but forgot in the Capsule one. Will do.

**Comment by ehelms** [comment]
Updated the tests including adding initial Foreman and Katello upgrade tests.

File: test/definitions/scenarios/capsule_upgrade_test.rb [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+        Procedures::Packages::Update,
+        Procedures::Service::Stop,
+        Procedures::Packages::Update,
+        Procedures::Installer::Run,
+      )
+    end
+
+    it 'composes all steps for Capsule on EL9' do
+      Scenarios::Satellite::Migrations.any_instance.stubs(:el_major_version).returns(9)
+      refute_scenario_has_step(scenario, Procedures::Packages::EnableModules)
```
Comment (evgeni) [inline test/definitions/scenarios/capsule_upgrade_test.rb:112]:
```suggestion
      refute_scenario_has_step(scenario, Procedures::Packages::EnableModules)
      refute_scenario_has_step(scenario, Procedures::Packages::SwitchModules)
```

File: test/definitions/scenarios/satellite_upgrade_test.rb [hunk 3]
```diff
@@ -1,39 +1,218 @@
...
+        Procedures::Packages::Update,
+        Procedures::Installer::Run,
+        Procedures::Installer::UpgradeRakeTask,
+      )
     end
 
     it 'composes all steps for Satellite on EL9' do
-      assume_feature_present(:satellite)
       Scenarios::Satellite::Migrations.any_instance.stubs(:el_major_version).returns(9)
       refute_scenario_has_step(scenario, Procedures::Packages::EnableModules)
```
Comment (evgeni) [inline test/definitions/scenarios/satellite_upgrade_test.rb:135]:
```suggestion
      refute_scenario_has_step(scenario, Procedures::Packages::EnableModules)
      refute_scenario_has_step(scenario, Procedures::Packages::SwitchModules)
```

**Review by evgeni — APPROVED** [review]

**Pull request merged**